process-migration-service/frontend/src/component/tabMigrationPlan/MigrationPlansBase.js

```diff
@@ -50,105 +40,49 @@ export default class MigrationPlansBase extends React.Component {
     });
   };
 
-  deletePlan = () => {
-    //need to create a temp variable "self" to store this, so I can invoke this inside axios call
-    const self = this;
-    const serviceUrl = BACKEND_URL + "/plans/" + this.state.deletePlanId;
-    axios
-      .delete(serviceUrl, {
-        headers: {
-          "Content-Type": "application/json"
-        }
-      })
-      .then(function() {
-        self.retrieveAllPlans();
-        self.hideDeleteDialog();
-      })
-      .catch(() => {
-        this.props.onError(
-          "Unable to delete the plan. Confirm the selected KIE Server is online"
-        );
-      });
+  deletePlan = async () => {
+    return PlanClient.delete(this.state.deletePlanId).then(() => {
+      this.retrieveAllPlans();
```

Comment:
delete dialog not appearing after clicking the button, should it be displayed explicitly?